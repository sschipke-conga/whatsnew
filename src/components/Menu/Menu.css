nav {
  display: flex;
  flex-direction: column;
  grid-area: menu;
  padding: 5px;
}

nav h2 {
  color: white;
}

nav span {
  color: yellow;
}

nav button {
  background-color: orange;
  border-radius: 20px;
  font-size: 1em;
  height: 5%;
  color: white;
  margin-top: 5px;
  outline: none;
}

nav button:hover {
  cursor: pointer;
}