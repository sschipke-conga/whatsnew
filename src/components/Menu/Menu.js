import React from 'react';
import './Menu.css'
// import entertainment from '../../data/entertainment'
// import health from '../../data/health'
// import science from '../../data/science'
// import technology from '../../data/technology'
// import local from '../../data/local'

const Menu = ({handleButton}) => (
  <nav>
    <h2>What's <span>New?</span></h2>
    <button name="health" type="button" onClick={(e) => handleButton(e)}>Health</button>
    <button name="entertainment" type="button" onClick={(e) => handleButton(e)}>Entertainment</button>
    <button name="science" type="button" onClick={(e) => handleButton(e)}>Science</button>
    <button name="technology" type="button" onClick={(e) => handleButton(e)}>Technology</button>
    <button name="local" type="button" onClick={(e) => handleButton(e)}>Local</button>
  </nav>
)

export default Menu;