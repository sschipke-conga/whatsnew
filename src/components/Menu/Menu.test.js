import React from 'react';
import { shallow } from 'enzyme';
import Menu from './Menu'

describe('Menu', () => {
  const mockHandleButton = jest.fn();
  let wrapper;
  beforeEach(() => {
    wrapper = shallow(<Menu 
      handleButton={mockHandleButton}
      />)
  });
  it('should match the snapshot', () =>{
    expect(wrapper).toMatchSnapshot()
  });
  it('should run handleButton function when the first button is clicked', () => {
    wrapper.find('button').at(0).simulate('click');
    expect(mockHandleButton).toHaveBeenCalled();
  });
  it('should run handleButton function when the second button is clicked', () => {
    wrapper.find('button').at(1).simulate('click');
    expect(mockHandleButton).toHaveBeenCalled();
  });
  it('should run handleButton function when the third button is clicked', () => {
    wrapper.find('button').at(2).simulate('click');
    expect(mockHandleButton).toHaveBeenCalled();
  });
  it('should run handleButton function when the fourth button is clicked', () => {
    wrapper.find('button').at(3).simulate('click');
    expect(mockHandleButton).toHaveBeenCalled();
  });
  it('should run handleButton function when the fifth button is clicked', () => {
    wrapper.find('button').at(4).simulate('click');
    expect(mockHandleButton).toHaveBeenCalled();
  })
})