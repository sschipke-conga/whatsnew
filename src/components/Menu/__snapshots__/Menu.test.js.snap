// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Menu should match the snapshot 1`] = `
<nav>
  <h2>
    What's 
    <span>
      New?
    </span>
  </h2>
  <button
    name="health"
    onClick={[Function]}
    type="button"
  >
    Health
  </button>
  <button
    name="entertainment"
    onClick={[Function]}
    type="button"
  >
    Entertainment
  </button>
  <button
    name="science"
    onClick={[Function]}
    type="button"
  >
    Science
  </button>
  <button
    name="technology"
    onClick={[Function]}
    type="button"
  >
    Technology
  </button>
  <button
    name="local"
    onClick={[Function]}
    type="button"
  >
    Local
  </button>
</nav>
`;
